stan/math/prim/fun/matrix_exp_action_handler.hpp

```diff
@@ -14,26 +16,59 @@ namespace math {
  * "Computing the Action of the Matrix Exponential,
  * with an Application to Exponential Integrators"
  * Read More: https://epubs.siam.org/doi/abs/10.1137/100788860
+ * See also:
+ * https://www.mathworks.com/matlabcentral/fileexchange/29576-matrix-exponential-times-a-vector
  *
  * Calculates exp(mat*t)*b, where mat & b are matrices,
  * and t is double.
  */
 class matrix_exp_action_handler {
   static constexpr int p_max = 8;
   static constexpr int m_max = 55;
-  static constexpr double tol = 1.1e-16;
```

Comment:
So I have established the following.
 - with `p_max` and `m_max` defined as `static constexpr int` as it is currently on develop, the expression tests fail with `undefined reference to stan::math::matrix_exp_action_handler::p_max`
    See https://jenkins.mc-stan.org/blue/organizations/jenkins/Math%20Pipeline/detail/PR-2619/22/pipeline/
- with `p_max` and `m_max` defined as `int` it passes the tests

It fails in the set_approx_order function below. This all seems very bizarre to me. Especially given that no one can replicate this locally and it might be an issue with the specific compiler.

cc @SteveBronder would you know what would be the cause of these issues?

Reply:
@SteveBronder would you know what is up in this case? Thanks!

Reply:
I think I found it: https://stackoverflow.com/a/8016853

I'm patching it on this branch.

Reply:
Update: multiple translation unit tests fail. Those check to make sure that we can link the math library within multiple translation units. It's clear that this approach fails for that test. I've updated the code to be simpler and not rely on static member variables.